src/psyclone/psyir/backend/fortran.py

```diff
@@ -192,6 +202,67 @@ def gen_declaration(self, symbol):
         result += "\n"
         return result
 
+    def container_node(self, node):
+        '''This method is called when a Container instance is found in
+        the PSyIR tree.
+
+        A container node is mapped to a module in the Fortran back end.
+
+        :param node: a Container PSyIR node.
+        :type node: :py:class:`psyclone.psyGen.Container`
+
+        :returns: the Fortran code as a string.
+        :rtype: str
+
+        :raises VisitorError: if the name attribute of the supplied \
```

Comment:
It also raises VisitorError if the Container has a child that is not a KernelSchedule.

Reply:
Well spotted.